File: frontend/src/types/revenueSplit.ts
export interface RevenueSplit {
  id: number
  title: string
  description: string
  totalAmount: number
  currency: string
  status: RevenueSplitStatus
  period: string
  members: SplitMember[]
  createdAt: string
  updatedAt: string
}

export type RevenueSplitStatus = 'DRAFT' | 'PENDING' | 'APPROVED' | 'DISTRIBUTED' | 'DISPUTED'

export interface SplitMember {
  id: number
  name: string
  email: string
  role: string
  percentage: number
  amount: number
  paymentStatus: 'PENDING' | 'PAID' | 'FAILED'
  paidAt?: string
}

export interface RevenueSplitSummary {
  totalSplits: number
  pendingAmount: number
  distributedAmount: number
  totalMembers: number
  averageSplit: number
}

export interface CreateRevenueSplitRequest {
  title: string
  description: string
  totalAmount: number
  currency: string
  period: string
  members: CreateSplitMemberRequest[]
}

export interface CreateSplitMemberRequest {
  name: string
  email: string
  role: string
  percentage: number
}

File: backend/onGo-domain/src/main/kotlin/com/ongo/domain/revenuesplit/RevenueSplit.kt
package com.ongo.domain.revenuesplit

import java.time.LocalDateTime

data class RevenueSplit(
    val id: Long = 0,
    val workspaceId: Long,
    val title: String,
    val description: String? = null,
    val totalAmount: Long = 0,
    val currency: String = "KRW",
    val status: String = "DRAFT",
    val period: String? = null,
    val createdAt: LocalDateTime = LocalDateTime.now(),
    val updatedAt: LocalDateTime = LocalDateTime.now(),
)
